### ISSUE TITLE
[Ubuntu18.04] Tests failed: System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test / *

### ISSUE THREAD

**Issue by karelz** [dotnet/corefx#27023]
Failed on newly added distro Ubuntu18.04

Affected tests:
* `GetAsync_AllowedSSLVersionDiffersFromServer_ThrowsException` (Tls11/Tls12 args)
* `GetAsync_DisallowTls10_AllowTls11_AllowTls12`

Test failure:
```
Assert.Throws() Failure
Expected: typeof(System.IO.IOException)
Actual:   typeof(System.Security.Authentication.AuthenticationException): A call to SSPI failed, see inner exception.

at System.Net.Security.SslState.StartSendAuthResetSignal(ProtocolToken message, AsyncProtocolRequest asyncRequest, ExceptionDispatchInfo exception) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 973
at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 820
at System.Net.Security.SslState.StartSendBlob(Byte[] incoming, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 791
at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 960
at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 863
at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 820
at System.Net.Security.SslState.StartSendBlob(Byte[] incoming, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 791
at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 960
at System.Net.Security.SslState.PartialFrameCallback(AsyncProtocolRequest asyncRequest) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 1094
--- End of stack trace from previous location where exception was thrown ---
at System.Net.Security.SslState.EndProcessAuthentication(IAsyncResult result) in /root/corefx-1372297/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 709
at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization) in /root/coreclr/src/mscorlib/src/System/Threading/Tasks/FutureFactory.cs:line 533
--- End of stack trace from previous location where exception was thrown ---
at System.Net.Test.Common.LoopbackServer.AcceptSocketAsync(Socket server, Func`5 funcAsync, Options options) in /root/corefx-1372297/src/Common/tests/System/Net/Http/LoopbackServer.cs:line 165
```

## History of failures

Day | Build | Details
-- | -- | --
2/10 | 20180210.01 | 3x - [link](https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20180210.01/workItem/System.Net.Http.Functional.Tests)
2/10 | 20180210.03 | 3x - [link](https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20180210.03/workItem/System.Net.Http.Functional.Tests)

**Comment by karelz** [comment]
Failed on newly added distro Ubuntu18.04 - @ianhays
@wfurt can you please take a look? Maybe it is just different default TLS settings than what we expect?

**Comment by danmosemsft** [comment]
Is SSPI relevant to Unix (should it be GSSAPI?)

Also exceptoin messages that end with ` see inner exception.` are not logging friendly. Better to append the inner exceptoin message.

**Comment by wfurt** [comment]
I've got repro. investigating. Ubuntu18 comes with openssl 1.1 and curl with gnutls.

**Comment by wfurt** [comment]
I submitted PR to disable failing tests. However this looks like product bug to me. 
The client sends TLS1.2 Client hello and that is reason why tests fail - they expect TLS/1.0 and failed negotiation. 

Packet capture from test run attached. 
[ubu18-tls.pcapng.zip](https://github.com/dotnet/corefx/files/1726120/ubu18-tls.pcapng.zip)

**Comment by wfurt** [comment]
It turned out this really is test bug. New OpenSSL sends Alert message about version mismatch and that is causing SSL Exception. Test only expects IO error as older openssl simply shuts down the socket. 

[test.cap.zip](https://github.com/dotnet/corefx/files/1746897/test.cap.zip)

**Comment by wfurt** [comment]
closed by #27352
